webpack/components/extensions/HostDetails/Tabs/ModuleStreamsTab/ModuleStreamsTab.js

```diff
@@ -0,0 +1,200 @@
+        return <><TableText wrapModifier="nowrap"> </TableText><CheckIcon color="green" /> {streamStatus}</>;
+      case (stream === 'Enabled' && streamStatus === 'Upgradable'):
+        return <LongArrowAltUpIcon color="blue"> {streamStatus}</LongArrowAltUpIcon>;
+      default:
+        return null;
+    }
+  };
+
+  EnabledIcon.propTypes = {
+    stream: PropTypes.string.isRequired,
+    streamStatus: PropTypes.string.isRequired,
+  };
+
+  return (
+    <div>
+      <div id="modulestreams-tab">
+        <TableWrapper
+          {...{
+          metadata,
+          emptyContentTitle,
+          emptyContentBody,
+          emptySearchTitle,
+          emptySearchBody,
+          searchQuery,
+          updateSearchQuery,
+          fetchItems,
+          status,
+          rowActions,
+        }}
+          additionalListeners={[hostId]}
+          fetchItems={fetchItems}
+          autocompleteEndpoint={`/hosts/${hostId}/module_streams/auto_complete_search`}
+          foremanApiAutoComplete
+          rowsCount={results?.length}
+          variant={TableVariant.compact}
+        >
+          <Thead>
+            <Tr>
+              {columnHeaders.map(col =>
+                <Th key={col}>{col}</Th>)}
+            </Tr>
+          </Thead>
+          <Tbody>
+            {results?.map((moduleStream, index) => {
+              const {
+                id,
+                status: moduleStreamStatus,
+                name,
+                stream,
+                installed_profiles: installedProfiles,
+                install_status: installStatus,
+                module_spec: moduleSpec,
+              } = moduleStream;
+
+              let color;
+              let labelVariant;
+              let disabledText = false;
+              /* eslint-disable max-len */
+              let streamStatus = moduleStreamStatus.charAt(0).toUpperCase() + moduleStreamStatus.slice(1);
+              streamStatus = streamStatus.replace('Unknown', 'Default');
+              if (streamStatus === 'Default') {
+                color = 'gray';
+                labelVariant = 'outline';
+              } if (streamStatus === 'Disabled') {
+                color = 'gray';
+                labelVariant = 'filled';
+              } if (streamStatus === 'Enabled') {
+                color = 'green';
+                labelVariant = 'filled';
+              }
+              let installedProfile = installedProfiles;
+              if (installedProfiles.length === 0) {
+                installedProfile = 'No profile installed';
+              }
+              if (installedProfiles.length > 0) {
+                installedProfile = installedProfiles.map(x => x[0].toUpperCase() + x.substr(1));
+                installedProfile = installedProfile.join(', ');
+              }
+              if (streamStatus === 'Disabled' || streamStatus === 'Default') {
+                disabledText = true;
+              }
```

Comment:
we should clean this up a bit, either with self-evoking functions/switches or through normal functions defined outside of the render.

Reply:
As a general rule, I like to pull out complicated logic from within the render to make the render look as clean as possible. When this is done across the app it makes it much easier to navigate later.